pkg/sink/store.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 The Witness Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sink
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/spiffe/go-spiffe/v2/spiffegrpc/grpccredentials"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	api "github.com/testifysec/archivist-api/pkg/api/archivist"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"io/ioutil"
+)
+
+type sink struct {
+	conn      *grpc.ClientConn
+	collector api.CollectorClient
+	archivist api.ArchivistClient
+}
+
+type Archivist interface {
+	GetBySubjectDigestRequest(ctx context.Context, algorithm, digest string) ([]string, error)
+}
+
+type Collector interface {
+	Store(attestation string, ctx context.Context) error
```

Comment:
Nitpicky, I know, but typically functions that take Contexts as a parameter should have them as the first parameter